releases.md

```diff
@@ -0,0 +1,25 @@
+# Release Guide
+
+This document explains the release process for the Python SDK. You can find the most recent version under [Github Releases](https://github.com/numaproj/numaflow-python/releases).
+
+
+### Before Release
+
+Before releasing a new SDK version, make sure to update all references from the old version to the new one. For example,
+the version in the root `pyproject.toml` should be updated (for [reference](https://github.com/numaproj/numaflow-python/commit/6a720e7c56121a45b94aa929c6b720312dd9340a)), 
```

Comment:
Looking at the link I saw 
```
Change release version to 0.7.0a0
Semantic versioning through poetry version preminor
```

Can we add more contexts on how to specify the <new-version>? what does `0.7.0a0` mean?

The goal is to make a non-python-expert (like myself) confident enough to execute a release.